### ISSUE TITLE
[Templates] Suggestion: Abilities > Show Off Cooldown and Almost Ready

### ISSUE THREAD

**Issue by amiyuy** [WeakAuras/WeakAuras2#797]
**Is your feature request related to a problem? Please describe.**
Suggestion for a new template type for Abilities -

Show the aura when the ability is off cooldown and when almost ready to be used (1.5/3/6/10 seconds depending on ability cooldown length).

**Describe the solution you'd like**
* Aura shows when ability is ready.
* Shows up desaturated when almost ready with (Inverse) Cooldown sweep and timer.

**Additional context**
Example:
```
dqKmiaqjbkgLaPoLaj3IkjSlbIggqQJHILHOEMavtJkX1eO02KsQVbKmoQKuNtkvMhvsP7rLKCqrslKk8qrvtukfxKkAJkjFukjnsPK4KIeZuq6McI2PK(PagkqCuPuAPkPEQQPkXvfe2kvsLVkqySujrNLkPQ9s(lvQblQCyKwmc9yPyYO0LPSze5ZIYOvItJQvtLu8APu1SLQBdWUv8BqdxelxONtvth66k12fuFhbJhqNhOwVuI9lsTyurNRktWjZqgu6EDYbjOcsgqRdqNvNvf9SoKGiQJgqKzfCWZBXuLHSoWaKi5nuvqRdoajsEdvf060PzbgbbYhupVZAX5A9MThHakgqmejSu003jSu005nWro0XiplZIEvuLrfDmYZYSOk6BV5oZglYB(cKqGro0ZSXI8MVajey09lqcbM1gLfS(2BU3dhLdDwJ4MePc4etVvCgkiRcrO6i4et3JuC719BIwyBWVjrIdu3D(uFX9lPUUOpuaMxfH6wyBq98qyNfsysNdKimlAEF6Cb9tSEpO0jUjr6lqcbgD)cKqGj1gKh20JgT)eR3RuaMUdZJCApDUGisXf9T3C7tSEVsbyYHUpX69uy9Qiu3VjAHTbbjAK4a1piodnu99yHTb1PBeQ7tSEVsbyQY0UGfuUOV9MBkrpmsmKqwto03dh1t6uWwuO(EWiplZIQONzJf5nFbsiWO7xGecmRnklyHcvLSk6yKNLzrv03dh1t6uWwuhbNy6EKIBVUFt0cBdcs0iXbQFqCgAO6(nrlSn43KiXbQ7oFQV4(Luxx03EZnLOhgjgsiRjh6BV5oZglYB(cKqGro0ZSXI8MVajey09lqcbM8qyNfsy03EZ9E4OCOhnA)jwVxPamDhMh50E6CbrKIl6rdaLp8jtV167XcBdQt3iupAaO8HpzUHjwKIWOUR1vPV9MBFI17vkato0TW2G65HWolKWKohiryw08(05c6Ny9EqPV9M7ObGYh(Kjh6(eR3RuaMQmTlybLlc13dg5zzwufHc1z5ExZEAqEytNoG1PnipSvI8SmlgsiRPQRGb0c15dAXH2DDw0ZHJQTtV9MLKQpXAW1uKdfrDV1jbcKpOEEN1IZ16SMNI7ifnruhnGidiq(G6ocukbc56uA1wPOJgqK8k4GR60mVEGqucnLQt9fJNTGQ2Ax07qkRk6(eRrfDa7oYvrOq9Mo17vrhWUJCveku3ZBXurhWUJCvekuOoAarY5Hds5daBq9uQpL24uNoCauKdhv09inIQOZUJuKdh3i4et3BnWH1RJGtm9boMqDau(OIo7osroCCJGtmDV1ahwVocoX0h4yc1jXhUFrfD2DKIC44gbNy6ERboSEDeCIPpWXekuhnGitEyhgvrvgvzuLSQmQgCvzu1fvzeQJgqKCEyhgvrvgvzuLSQmQgCvzu1fvzeQJgqKX5qj2nw5qhnGiZk4GR60mVEGqucnLQt9OLXHtbCIPZBGJEdSdJQOkJQmQswvgvdUQmQ6IQmc1BGWolKWumGOZHsSBSYHoAarYRGdEElMQKDrhnGizqG8b1Z7SwCUwVfieGQGwhnGiJEW41jWzXfvBTl60EYcvvgD0aIK1dMeD6gHYHoAarYohkXUXkI6Cw9VajKoxasKsNlpe2zHeM05OdB6CP2ba9y6CRnklyDpYBAp42SJuKdhruNuKcWogPivvgD60SaJRIuawbCIP)8RD6SQJgqK8k4G6RI8wsNRTBkBKI005kf9nNvV0wkOC1eD9GBPXvQZNWw0BYHoAarMvWb1xf5TKoxB3u2ifPPZvk60gKdhVk6EKgrveQtIpC)Ikc15dhvrOq9g4WYroC8QOkJk6nlwt7vrhJ8SmlQkz9W0iNsSB6(fib9W0(2uLrOEZcDYmVkQYOIEyAFBYHEseMyreSoR5P4osrtOqHc1BGWolKWih6ObezYdhKYha2G6PuFkTXPq9WQY4czgHs
```

**Comment by emptyrivers** [comment]
Ooh, I like this one.

**Comment by amiyuy** [comment]
For all of my personal auras like this I use Action Usable for the Show Off Cooldown trigger instead of Cooldown Progress, but Cooldown Progress seems to be the preferred trigger so I tried to use that in the first suggestion.  Unfortunately for spells with stacks/charges it's a little more complicated and it seems Action Usable is necessary.  There might be a better way, but I'm not sure what it is.

Triggers
1. Custom trigger: return[t1] or [t2]
2. Dynamic Info: Trigger 3
2. Trigger 1: Action Usable
3. Trigger 2: Cooldown Progress > Remaining Time & On Cooldown
4. Trigger 3: Cooldown Progress > Always

Conditions
1. If Trigger 1 > Stacks >= 2 > Glow
2. If Trigger 2 > Stacks = 0 > Desaturate

Example with Guerrilla Tactics talent and Wildfire Bomb as Survival Hunter:
```
dC0zkaqjjP0OaQYPaQQBjjH2LKKAyakhtuTmuPNjezAcPUgG02ef5BsssNdq17KKiZtscUhGW(eIQoiuvleQ8qjvteq0fLeBeQYhLKIrkjvDsjLwPquMPqYnff1ov4NIsdfOCujfAPcHNQAQsCvrbBvsI6RskATssI3kPG7kjv2l1FrfdgkomrlgGhlIjluxgzZq4ZqA0kQtJYQfIkVwuOzlv3wr2Ts)g0WfPLt45cMoPRlLTJQ8DiA8qPZJQA9av2pq25U4Z8GlWYeqZuKI2p4ZTQRQvDoW8N8J9JDXhTdrAa(kHvZXdU6zGJ8iNRFce2JHi3cHvRSsaDk2485NfbIaPEamF5MmdfGbzR6xVYOujc)KwqHtfcRMH0yPs(T0yPs(YeLb3Gl(bvkuxS6JGTSWSlw9zltDXQvFvWqrjrWfpYDXxfmuus4IFlqCqPLeSKpdrMDnoFuAjbl5ZqKzx)Wmez2ncsgZ3VfioTLPgNFmbOHarHFk5x9vIcm8Yqu(k)uYpOsTf8dnaepA13qGGH1NtD8FHZl4hrXhqdbIpdrMD9dZqKzx8tugpYN4rR6JFIYGlimziib0j)qdaXJwfmbHGH1)GvjQO8fKS)uQ3hYjYVMSEmcMGaHbhCcGFlqCcPuVpKtKX5hsPEV24Glw9Blt9t7s(KWVTepAvFztH(HuQ3hYjYdUaf4rdCG63cehjGauaiZWyY48x5efCXQv)2QcgkkjCXhLwsWs(mez21pmdrMDJGKX8TA1dUU4RcgkkjCXp0aq8Ovbtqiyy9pyvIkkFLFk5huP2c(TaXrciafaYmmMmo)2s8Ov9Lnf63ceN2YuJZxqts2YwuoWusivOWVkae(O0scwYNHiZU(HziYSBDiShdrU(TLP(PDjFs4liz)PuVpKtKFnz9yembbcdo4eaFbnjzlBr9ZKFObG4rR(gcemS(CQJ)lCEb)ik(TaXrqts2YwuJZVfioHuQ3hYjY48jE0Q(1HWEme5ccdyciQGcbqyaVNs9o473cehuAjbl5ZqKzxJZpKs9(qorEWfOapAGduR(Tvfmuus4IvREejx8vbdfLeU4R8tj)Gk1wWp0aq8OvFdbcgwFo1X)foVGFef)wG4GsljyjFgIm7AC(epAv)6qypgICbHbmbevqHaimG3tPEh89BbItBzQX5liz)PuVpKtKFnz9yembbcdo4ea)wG4esPEFiNiJZVTepAvFztH(TLP(PDjFs4hAaiE0QGjiemS(hSkrfLFlqCKacqbGmdJjJZhLwsWs(mez21pmdrIFhPKFWpKs9(qorEWfOapAGduR(Tvfmuus4IpkTKGL8ziYSRFygIe)osj)GvR(XSqKRTjkJh5N0ckCYpPfu4urWqrjruquwIpI2MOm4cEQGHIscWhuKbcdimGWacJG0MybHrfmuusud5vhimqbimQGHIsIAGB1vLafz0g7ltugpQiyOOKiZWyYJiz1xjSAELvcOtXgNFgjkf)qkLOrivg0a8LlBsQm46IFqLc1f)4MqQm4Yr5Ns(bkbUXbFLFk5VWLS6pjzRl(XnHuzWLJYpL8ducCJd(k)uYFHlz1hbBzHzx8JBcPYGlhLFk5hOe4gh8v(PK)cxYQvFLWQ51HRkz7eTQFTJxlqwXxjSAoyq2Q(4YwkzZCe1wnvFXxjSkx8GRIxxIg8ZMHsu1oQ4ptm0z1JmfTFhkJDXpKsjU4p16kZfRw9t6YqWf)PwxzUy1QFlqCwIhjma)adCKl(tTUYCXQvR(kHv5whUQKTt0Q(1oETazf)yki1MqQKb4Rewnhp4Q(4jyGdeMASjJfsvcctP4Rewnhp4Q41LOb)SzOevTJk(kHv5IhCvF8emWbctn2KXcPkbHPu8vcRYToSdfU4rUh5EW1JCpIKh5EeTh5w9fekdUf(PKplbU(yZIarGupaMFcSdfU4rUh5EW1JCpIKh5EeTh5w9vcRYfp4QNboYdUr7RewLlyq2Q(1RmkvIWxjSkx)Qn1xjSAUF1g8rYI1zpYu0(YE6S0JCFWbHtEamFztHgNVsyvUvwjGofBa(Sy)pdrcctweiaHPoe2JHixqyKBmim43NGbfeMiizmFqyQIGWuFwkqPGFqzjzKpq2esLbxdWhHqorxfsv6rUVCtMHc8eYjQWpL8F9iQuz4hOocWGSv9RxzuQeHFJf7Zag4aRMvLQH28iZNT8irGmoFwcCnoF2QKyL9Qmjcm46bW9tGBmtzWn4Ih5U4Nmtjz0fFvWqrjHh5(Wu)ipq4Zt2BKpxFEsbtcOt(bvMKXGv)Kz5Isbx8i3fFEYEJmo)ubmLekFF0oePvRw9GRl(jZusgDXxfmuus4bxFyQpqae(8K9g5dmFEsbtcOt(jZsbkfS6Nmlxuk4Ih5U4Zt2BKX5NkGPKq57htbP2esLSA1Qv)eiShdrUgNVsy186Wou4Ih5EK7bxpY9isEK7r0EKB1QpppYJMBUvBa
```

**Comment by Stanzilla** [comment]
This is getting close to what I meant when I said templates could be transformed to a wizard style flow where you pick your preferences and spells and it generates a group for you

**Comment by Vrakfall** [comment]
I'm currently working on something quite similar: For the spell with debuff `Agony`, for example, I have an aura that appears 3 seconds before the spells disappears from the target. This is all done with re-usable code so the special animation I coded can be re-used for spells that don't need those 3 seconds, e.g..

As suggested in #809, I first need to create an add-on to gather the re-usable part of my code. Once that's done, I'll show what I've got, that might interest you.
